atom/feature_engineering.py

```diff
@@ -988,12 +1014,68 @@ def check_y():
+            # initialization_params catches all params requied for zoofs algos
+            initialization_params = {"pso": ['n_iteration', 'timeout', 'population_size', 'minimize', 'c1', 'c2', 'w'],
+                                     "gwo": ['n_iteration', 'timeout', 'population_size', 'minimize', 'method'],
+                                     "dfo": ['n_iteration', 'timeout', 'population_size', 'minimize', 'method'],
+                                     "geneo": ['n_iteration', 'timeout', 'population_size', 'minimize', 'selective_pressure',
+                                               'elitism', 'mutation_rate'],
+                                     "hho": ['n_iteration', 'timeout', 'population_size', 'minimize', 'beta']}[self.strategy.lower()]
+            initialization_params_from_kwargs = {
+                k: v for k, v in self.kwargs.items() if k in initialization_params}
+            params_for_objective_function = {k: v for k, v in self.kwargs.items(
+            ) if k not in initialization_params+['objective_function']}
+
+            
+            if self.kwargs.get("objective_function"): # zoofs-native objective_function choice
+                objective_function = self.kwargs['objective_function']
+
+            elif self.kwargs.get("scoring"): # atom-native scoring method can be chosen 
+                params_for_objective_function = {
+                    "scorer": get_custom_scorer(self.kwargs["scoring"])}
+                objective_function = custom_function_for_scorer
+
+            else: # default scoring method is chosen if nothing is provided
+                params_for_objective_function = {"scorer": get_custom_scorer("neg_log_loss") if hasattr(
+                    self._solver, "predict_proba") else get_custom_scorer("neg_mean_squared_error")}
+                objective_function = custom_function_for_scorer
+
+            if 'minimize' not in initialization_params_from_kwargs.keys():
+                initialization_params_from_kwargs['minimize'] = False
+
+            self.algo = algo_mapper[self.strategy.lower()](objective_function=objective_function,
+                                                           logger=self.logger,
+                                                           **initialization_params_from_kwargs,
+                                                           **params_for_objective_function)
+
+            if ("X_valid" in self.kwargs.keys()) and ("y_valid" in self.kwargs.keys()): # catching X_valid if provided
+                X_valid, y_valid = self._prepare_input(
+                    self.kwargs["X_valid"], self.kwargs["y_valid"])
+            else:
+                X_valid, y_valid = X, y
+
+            self.algo.fit(model=self._solver,
+                          X_train=X,
+                          y_train=y,
+                          X_valid=X_valid,
+                          y_valid=y_valid,
+                          verbose=False if self.verbose == 0 else True)
```

Comment:
Should be `if self.verbose > 1 else`. ATOM prints the output of external libraries only for verbose=2.

Reply:
thanks for pointing it out, will make the change